core/src/main/java/io/questdb/std/Hash.java

```diff
@@ -98,6 +101,14 @@ public static int spread(int h) {
         return (h ^ (h >>> 16)) & SPREAD_HASH_BITS;
     }
 
+    /**
+     * Assumes that the string contains ASCII chars only. Strings with non-ASCII chars
+     * are also supported, yet with higher chances of hash code collisions.
+     */
+    public static long xxHash64(String str) {
+        return xxHash64(0, str.length(), 0, stringUtf8MemoryAccessor.get().of(str));
```

Comment:
Thread locals are usually quite expensive.
Are you sure you benched this code path and there is still a benefit?

Reply:
Yup, it's benched. `DirectByteCharSequence*HashMap` now cache stored hash codes, so string hashe codes are only calculated once. On the hot path we only do look-ups for `DirectByteCharSequence`s, so that's what we optimize for.

Reply:
Please move `DirectByteCharSequenceHashMap` into ILP package and make it package private, seems that it cannot be used generically and `std` is not a good place for it

Reply:
Also since it's not for the hot path I would move ThreadLocal out of `Hash` class into `DirectByteCharSequenceHashMap`

Reply:
Why you did not want to have a version with inlined accessor for Strings and regular CharSequence? It could be useful in other hash maps since it's faster.

Reply:
> Why you did not want to have a version with inlined accessor for Strings and regular CharSequence? It could be useful in other hash maps since it's faster.

I didn't want to end up with N versions of the same complex hash function. JIT compiler should be able to inline the accessor.

Reply:
Moved the maps to `.cutlass.line.tcp` in 7bc20de53dfc67ead372529b6de385479e083e01 as well as the thread-local and `StringUtf8MemoryAccessor`. Had to keep `DirectByteCharSequenceIntHashMap` public since it's used in the JMH benchmark.

Reply:
One more follow-up: removed the thread local completely in c36de175813fd349c063c77d82fa2a7f3897c2c9